syntax = "proto3";
option go_package = "trustedStorage/grpsServer";
import "google/protobuf/empty.proto";

message CreateTx {
  bytes senderAddress = 1;
  bytes data = 2;
  bytes pubKey = 3;
  bytes signature = 4;
  bytes documentBytes = 5;
}

message CreateResponse {
  string response = 1;
}

message User {
  bytes senderAddress = 1;
}

message BlockHeader {
  uint64 version = 1;
  bytes HashPrevBlock = 2;
  //HashMerkleRoot []byte
  int64 Time = 3;
}


message Block {
  //MacigNo
  uint64 blocksize = 1;
  BlockHeader blockheader = 2;
  uint64 TxCounter = 3;
  repeated CreateTx transactions = 4;
}

service Invoicer {
  rpc AddTxToBlockchain(CreateTx) returns (CreateResponse);
  rpc GetLastBlock(google.protobuf.Empty) returns (CreateResponse);
  rpc GetBlockchain(google.protobuf.Empty) returns (CreateResponse);
  rpc GetUserTxHistory(User) returns (CreateResponse);
}